server/src/main/java/org/elasticsearch/action/admin/indices/resolve/TransportResolveClusterAction.java

```diff
@@ -253,6 +253,9 @@ private boolean notConnectedError(Exception e) {
         if (e instanceof ConnectTransportException || e instanceof NoSuchRemoteClusterException) {
             return true;
         }
+        if (e instanceof IllegalStateException && e.getMessage().contains("Unable to open any connections")) {
```

Comment:
having to compare the message to define the behaviour is really delicate. Are there other illegal state that are thrown? Do we need a different exception? I think we should look for a better way to do this.

Reply:
That would take some discussion with the distributed team as who knows what other code depends on that model. If they change the error message, then this test will break and we can change it. And even if we don't, the issue is not serious (see below).

> This looks like a potential bug and we should mark it as such and backport the fix?

We could, but it's not serious. It's a cosmetic issue at best. The difference in behavior between before and after is whether an error message is included in addition to saying connected=false.

Here's the result now:

```
{
  "remote1": {
    "connected": false,
    "skip_unavailable": false,
  }
}
```

Here's the result without the additional handler added in this PR:

```
{
  "remote1": {
    "connected": false,
    "skip_unavailable": false,
    "error": "Unable to open any connections to [remote1]"
  }
}
```

The handlers I added to the code was to avoid putting error messages around connectivity, because `connected: false` is all the user really needs to see. If an unexpected error occurs, I default to `connected: false` to be safe and display the error message.